fog/ingest/enclave/impl/tests/tx_processing.rs

```diff
@@ -94,7 +94,7 @@ fn test_ingest_enclave(logger: Logger) {
             // Check that the tx row ciphertexts have the right size (but only at block
             // version max)
             if block_version == BlockVersion::MAX {
-                const EXPECTED_PAYLOAD_SIZE: usize = 237; // The observed tx_row.payload size
+                const EXPECTED_PAYLOAD_SIZE: usize = 239; // The observed tx_row.payload size
```

Comment:
@remoun for some reason after I changed the TxOutRecord to use `oneof`, the size of this seems to have increased. i'm going to tinker around with this some more and make sure, but if true that's a reason not to use oneof

Reply:
i guess it's probably fine either way, we're still at 239 + 1 + 16 = 256 bytes exactly

Reply:
and circuit oram is coming in this release so the 256 bytes thing probably won't be as critical

Reply:
If the size increased could this potentially indicate the change is not backwards compatible?

Reply:
Not sure. We should be able to test it using CD

Reply:
Seems important...

Reply:
let me explain, there were two separate changes:

1. Make TxOut use a `oneof` field where previously we had a masked amount message
2. Make TxOutRecord (used in fog) use a `oneof` field where previously we had a `bytes` type

(1) did not cause a change in size here. But (2) did, and I'm not sure why.

I'm not worried about an incompatibility in the blockchain part, because that did not affect the size, and we went through CD with that. Would like to better understand what actually caused the size increase at step (2), maybe i screwed up the protobuf annotation or something.

Reply:
I managed to figure this out:

```
    /// A v1 masked amount's token id bytes.
    /// This is used when the masked amount in the original TxOut has version 1
    /// Note: This tag must match the historical tag used for masked token id
    /// bytes
    #[prost(bytes, tag = "10")]
    V1(Vec<u8>),

    /// A v2 masked amount's token id bytes
    /// This is used when the masked amount in the original TxOut has version 2
    #[prost(message, tag = "11")]
    V2(Vec<u8>),
}
```

The above is wrong, I marked a bytes as a message.

I found this using protobuf inspector:

https://github.com/mildsunrise/protobuf-inspector

```
(base) chris@chris-ThinkPad-X1-Extreme-Gen-3:~/mobilecoinofficial$ protobuf_inspector < foo_proto_old
root:
    2 <64bit> = 0x5413595A7527A94F / 6058284168790911311 / 1.0332323e+97
    3 <chunk> = bytes (32)
        0000   28 DF 86 34 4C CB E0 B9 9D E6 CF DD B4 FB A4 0D 69 76 A9 D6 31 C2 0E 7F  (..4L...........iv..1...
        0018   F5 14 9A 4E CF 42 8D 46                                                  ...N.B.F
    4 <chunk> = bytes (32)
        0000   A4 B4 2C CA BD 98 32 1F 97 6F 95 7D 5C F2 02 6D 8F 31 B5 2B A3 C2 CC 25  ..,...2..o.}\..m.1.+...%
        0018   0F 82 2D A5 FC 3B C5 25                                                  ..-..;.%
    5 <64bit> = 0x000000000000006D / 109 / 5.3853155e-322
    6 <64bit> = 0x0000000000000001 / 1 / 4.9406565e-324
    7 <64bit> = 0x000000000000000A / 10 / 4.9406565e-323
    8 <32bit> = 0xEFC5EAB6 / -272241994 / -1.22505e+29
    9 <chunk> = bytes (66)
        0000   6D BB 45 75 D2 37 D0 7D A5 8F B5 DB 83 E0 86 50 DE 03 C6 E6 35 DB 09 86  m.Eu.7.}.......P....5...
        0018   B3 7F 16 E3 C4 31 68 EB 2B 50 0D E3 1B 69 74 2A 37 CE 29 0C 79 51 D8 FD  .....1h.+P...it*7.).yQ..
        0030   DF 85 00 16 C0 C8 92 2B DD CD 51 5A 5D 64 04 B8 5A EA                    .......+..QZ]d..Z.
    10 <chunk> = bytes (8)
        0000   88 D4 B1 3A D4 0D 88 F3                                                  ...:....

(base) chris@chris-ThinkPad-X1-Extreme-Gen-3:~/mobilecoinofficial$ protobuf_inspector < foo_proto_new
root:
    2 <64bit> = 0x95CDAD403242D4A3 / -7652269700599851869 / -1.1831762e-203
    3 <chunk> = bytes (32)
        0000   28 DF 86 34 4C CB E0 B9 9D E6 CF DD B4 FB A4 0D 69 76 A9 D6 31 C2 0E 7F  (..4L...........iv..1...
        0018   F5 14 9A 4E CF 42 8D 46                                                  ...N.B.F
    4 <chunk> = bytes (32)
        0000   A4 B4 2C CA BD 98 32 1F 97 6F 95 7D 5C F2 02 6D 8F 31 B5 2B A3 C2 CC 25  ..,...2..o.}\..m.1.+...%
        0018   0F 82 2D A5 FC 3B C5 25                                                  ..-..;.%
    5 <64bit> = 0x000000000000006D / 109 / 5.3853155e-322
    6 <64bit> = 0x0000000000000001 / 1 / 4.9406565e-324
    7 <64bit> = 0x000000000000000A / 10 / 4.9406565e-323
    8 <32bit> = 0xD16902BE / -781647170 / -6.25483e+10
    9 <chunk> = bytes (66)
        0000   6D BB 45 75 D2 37 D0 7D A5 8F B5 DB 83 E0 86 50 DE 03 C6 E6 35 DB 09 86  m.Eu.7.}.......P....5...
        0018   B3 7F 16 E3 C4 31 68 EB 2B 50 0D E3 1B 69 74 2A 37 CE 29 0C 79 51 D8 FD  .....1h.+P...it*7.).yQ..
        0030   DF 85 00 16 C0 C8 92 2B DD CD 51 5A 5D 64 04 B8 5A EA                    .......+..QZ]d..Z.
    11 <chunk> = message:
        1 <chunk> = bytes (8)
            0000   2B 79 59 A3 62 8B B3 9F                                                  +yY.b...

```